be/src/vec/exprs/table_function/vexplode_json_object.cpp

```diff
@@ -0,0 +1,159 @@
+
+#include <glog/logging.h>
+
+#include <ostream>
+#include <vector>
+
+#include "common/status.h"
+#include "vec/columns/column.h"
+#include "vec/common/string_ref.h"
+#include "vec/core/block.h"
+#include "vec/core/column_with_type_and_name.h"
+#include "vec/exprs/vexpr.h"
+#include "vec/exprs/vexpr_context.h"
+
+namespace doris::vectorized {
+
+VExplodeJsonObjectTableFunction::VExplodeJsonObjectTableFunction() {
+    _fn_name = "vexplode_json_object";
+}
+
+Status VExplodeJsonObjectTableFunction::process_init(Block* block, RuntimeState* state) {
+    CHECK(_expr_context->root()->children().size() == 1)
+            << "VExplodeJsonObjectTableFunction only support 1 child but has "
+            << _expr_context->root()->children().size();
+
+    int text_column_idx = -1;
+    RETURN_IF_ERROR(_expr_context->root()->children()[0]->execute(_expr_context.get(), block,
+                                                                  &text_column_idx));
+
+    _json_object_column = block->get_by_position(text_column_idx).column;
+    return Status::OK();
+}
+
+void VExplodeJsonObjectTableFunction::process_row(size_t row_idx) {
+    TableFunction::process_row(row_idx);
+
+    StringRef text = _json_object_column->get_data_at(row_idx);
+    if (text.data != nullptr) {
+        JsonbDocument* doc = JsonbDocument::createDocument(text.data, text.size);
+        if (UNLIKELY(!doc || !doc->getValue())) {
+            // error jsonb, put null into output, cur_size = 0 , we will insert_default
+            return;
+        }
+        // value is NOT necessary to be deleted since JsonbValue will not allocate memory
+        JsonbValue* value = doc->getValue();
+        auto writer = std::make_unique<JsonbWriter>();
+        if (value->isObject()) {
+            _cur_size = value->length();
+            ObjectVal* obj = (ObjectVal*)value;
+            _object_pairs.first =
+                    ColumnNullable::create(ColumnString::create(), ColumnUInt8::create());
+            _object_pairs.second =
+                    ColumnNullable::create(ColumnString::create(), ColumnUInt8::create());
+            _object_pairs.first->reserve(_cur_size);
+            _object_pairs.second->reserve(_cur_size);
+            for (auto it = obj->begin(); it != obj->end(); ++it) {
+                _object_pairs.first->insert_data(it->getKeyStr(), it->klen());
+                writer->reset();
+                writer->writeValue(it->value());
+                if (it->value()->isNull()) {
+                    _object_pairs.second->insert_default();
+                } else {
+                    const std::string_view& jsonb_value = std::string_view(
+                            writer->getOutput()->getBuffer(), writer->getOutput()->getSize());
+                    _object_pairs.second->insert_data(jsonb_value.data(), jsonb_value.size());
+                }
+            }
+        }
+        // we do not support other json type except object
```

Comment:
should we add test cases for other types?

Reply:
in qt_order_select_explode_json_object_out case , we treated json other type as NULL for output